### DISCUSSION TITLE
Embedded Images Fail to Appear (all of a sudden, was working)

### DISCUSSION THREAD

**Discussion by 4dsolutions** [community/community#83692] [New to GitHub]
I hope I'm posting in the right place. I'm perplexed because my embedded images are no longer showing as of Jan 1 2024 in my many repos. 

Consider:  
https://github.com/4dsolutions/m4w/blob/main/Polyhedrons.ipynb
https://github.com/4dsolutions/MartianMath/blob/master/mm5.ipynb

The main pix have been coming from Flickr and all has been smooth. At the very top left, are "broken image" icons next to 'Open in Colab' and  'nbviewer'. Both of these pull up the very same Jupyter Notebook in different hosting enviros (colab and nbviewer) and these work fine as they always have, with the images all showing up. 

Compare:
https://nbviewer.org/github/4dsolutions/m4w/blob/main/Polyhedrons.ipynb
https://nbviewer.org/github/4dsolutions/MartianMath/blob/master/mm5.ipynb

The images also all show up locally before I push to github. I haven't made any significant changes to trigger such a global change in behavior. I've tested on different computers with different browsers. Note these pix are coming from outside Github, but that hasn't been a problem until today.

**Comment by afidyoga** [comment]
If you are experiencing issues on GitHub where embedded images don't consistently appear or suddenly start working, there are several steps you can try to address the problem:

Refresh the Page:
Refresh the GitHub page. Sometimes, temporary issues can be resolved by refreshing the page.

Clear Cache:
Try clearing your browser cache. Cache that is sometimes expired or corrupted can cause display issues, such as images not appearing.

Check Image URL:
Ensure that the URL used to embed the image is correct. Use the full link pointing to the correct location of the image in the repository or an external site.

Image Format and Extension:
Make sure the embedded image has a supported format and extension. GitHub supports many common image formats such as PNG, JPEG, GIF, etc.

Check Internet Connection:
Ensure your internet connection is stable. If the connection is interrupted or slow, images may not load properly.

Use Absolute Links:
If you are embedding images from a different repository, consider using absolute links (like https://github.com/username/repository/path/to/image.jpg) instead of relative links.

GitHub Status:
Check the GitHub status on the [GitHub Status Page](https://www.githubstatus.com/). There might be a common issue on the platform affecting image availability.

Check Repository Configuration:
Ensure there are no access restrictions or specific configurations in the repository that could hinder image retrieval.

Contact GitHub Support:
If all the above steps fail and the issue persists, consider reaching out to GitHub's technical support for further assistance.

If you have additional information or error messages that appear, it can help in determining the root cause of the issue.

**Reply by 4dsolutions** [reply]
Given the nbviewer.org and colab.research.google.com Jupyter Notebook servers are still correctly rendering the embedded images in question, in the very same files pulled from Github, I can only conclude that Github has somehow inadvertently dumbed down its own JN server or has otherwise blocked 3rd party pictures. I doubt any action will be taken so long as it's "just me" so I'm on the lookout for any corroborating reports.

**Reply by marwan116** [reply]
I am facing the same issue with notebooks no longer rendering images that were rendering fine a few days ago

**Reply by 4dsolutions** [reply]
I don't think these imported (off site) images are getting handed off to camo, the image caching service. Searching those anonymized URLs directly gives Bad Signature error. It might be an intentional policy change giving we're here at the year flipover, I just haven't found where it's spelled out. Thankfully nbviewer is still doing the job. 

I have a lot of somewhat fancy embedded HTML suggested by Flickr (example appended), and that was all working just fine, never an issue. I've substituted the simplest Markdown `![name](URL)` syntax in some cells for testing, but no dice: still not downloading. Yet when I click on the broken glass icon, it takes me right there (to the intended picture on Flickr).

Example of what used to work (doesn't since New Years). Hey it works here!

<a data-flickr-embed="true" href="https://blahblah" title="Business Accelerator Building"><img src="https://live.staticflickr.com/1584/23636692173_103b411737_w.jpg" width="400" height="300" alt="Business Accelerator Building"></a>
<script async src="//embedr.flickr.com/assets/client-code.js" charset="utf-8"></script>

**Comment by andrewyang94** [comment]
+1 I'm facing the same issue here. My personal investigation seems to suggest that what we're seeing is associated with URL-linked images (rather than images that are present in your repo that you link via an absolute or relative path), and specifically within Jupyter Notebooks. Note that URL-linked images will work in a `README.md`, for example, but not when previewing a Jupyter Notebook via Github's web preview feature.

Therefore, I agree this has something to do with either camo or the JN rendering backend that GitHub uses. I've also tried to flush the cache for camo via https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls#removing-an-image-from-camos-cache, but still nothing.

If I attempt a `curl` for the proxied image, I get the following:

```
(atoti) aya@aya atoti % curl -I https://camo.githubusercontent.com/aeeb9ca5745fe59343ed8569b77c56f82cf01c61cada2ea56bc992af841ced61/68747470733a2f2f646174612e61746f74692e696f2f61746f74692d61636164656d792f696d616765732f696e74726f64756374696f6e2f305f41746f74695f6172636869746563747572652e706e67
HTTP/2 403 
cache-control: no-cache, no-store, private, must-revalidate
content-security-policy: default-src 'none'; img-src data:; style-src 'unsafe-inline'
content-type: text/plain; charset=utf-8
server: github-camo (c20f7b3f)
strict-transport-security: max-age=31536000; includeSubDomains
x-content-type-options: nosniff
x-frame-options: deny
x-xss-protection: 1; mode=block
x-github-request-id: 6B05:20FB8B:1D511DC:20E6FCF:66D76766
accept-ranges: bytes
date: Tue, 03 Sep 2024 19:45:53 GMT
via: 1.1 varnish
x-served-by: cache-ewr-kewr1740062-EWR
x-cache: MISS
x-cache-hits: 0
x-timer: S1725392754.936581,VS0,VE9
x-fastly-request-id: b51ac1247a17eec67677527145434a7d833ac22e
timing-allow-origin: https://github.com
content-length: 14
```

Which results in a `403` and I also see `style-src 'unsafe-inline'`. So I also think that there is some content security policy violation going on or something.

**Comment by 4dsolutions** [comment]
>
> +1 I'm facing the same issue here. My personal investigation seems to
> suggest that what we're seeing is associated with URL-linked images (rather
> than images that are present in your repo that you link via a absolute or
> relative path), and specifically within Jupyter Notebooks. Note that
> URL-linked images will work in a README.md, for example, but not when
> previewing a Jupyter Notebook via Github's web preview feature.



As of right now, on Sep 3 1:10 PM PDT, it's working again for me.

A lot of my Jupyter Notebooks pull from my Flickr Photostream by embedded
URL, using fancy syntax like:

<a data-flickr-embed="true" href="
https://www.flickr.com/photos/kirbyurner/479693617/" title="Models"><img
src="https://live.staticflickr.com/214/479693617_fdab22ae6e_o.jpg"
width="417" height="331" alt="Models"/></a><script async src="//
embedr.flickr.com/assets/client-code.js" charset="utf-8"></script>

at:

https://github.com/4dsolutions/School_of_Tomorrow/blob/master/School_of_Tomorrow.ipynb

The same page will display images pulled in using exclamation point syntax
MarkDown, from the local repo, but not the embedded stuff.

Hence the banner graphic at the top of this particular Jupyter Notebook,
reminding teachers who use this repo to clone it locally or use nbviewer or
Google colab.

But as I said, right now it's back to working, after many days of not.

I'm of the view it's not something a Github enduser can do anything about
except report on.

Kirby

**Reply by andrewyang94** [reply]
Woahhh... I don't know what happened but I input your HTML into my Jupyter Notebook and did a `git push`, and now all of my embedded images in my Jupyter Notebooks in GitHub preview are now all rendering properly when they literally were not working only 5 minutes ago. Not sure if your snippet cleared up something or GH is monitoring this thread but I guess I'll take the win and see myself out 🎩 — but thank you for sharing and opening this thread. Seems like there is a bit of uncertainty as to this issue, but I'm glad it's working (for now).

https://github.com/atoti/atoti/blob/main/01-atoti-academy/introduction-to-atoti/main.ipynb

**Comment by joshuapinter** [comment]
Just coming here to say that ours was not working either so I went investigating, wound up here and by the time I went back and refreshed, things were working again. So, it does happen from time to time. Check in a bit before wasting time debugging.

**Comment by LadnarofLadner** [comment]
Me three? I am 99% scientist and 1% guy-who-maintains-his-own-github-page-as-a-peer-un-reviewed-publishing-of-first-rate-scholarship, so, I don't have time to go back and do work that was already done? Why so glitchy on the images? I guess I can contact github tech support? maybe anybody finding my complaint, here, could go to my webpage and tell me if the pictures are loading? I can click on the broken image icon, rather, right click, and tell it to re-load. It works about 10% of the time. But for 20 or 50 photos? Webpages should just load. The 1990s solved that problem when they abandoned the squirrel chatter hard drive and the shrieking 56k baud modem. OK, please let me know if you can access : https://ladnarofladner.github.io/VisionResearch.html without hassles-upon-hassles.

**Comment by fengfeng-222** [comment]
Consider another way to host your image, with free and larger storage. Check it out here:
1.[imgbb.com/](https://imgbb.com/)
2.[postimages.org/](https://postimages.org/)
3.[pages.edgeone.ai/use-cases/image-to-url](https://pages.edgeone.ai/use-cases/image-to-url)
